### PR TITLE
Do no-op if mountpoint is shared

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

**Comment by StrongMonkey** [comment]
```
/* 36 35 98:0 /mnt1 /mnt2 rw,noatime master:1 - ext3 /dev/root rw,errors=continue
	   (1)(2)(3)   (4)   (5)      (6)      (7)   (8) (9)   (10)         (11)

	   (1) mount ID:  unique identifier of the mount (may be reused after umount)
	   (2) parent ID:  ID of parent (or of self for the top of the mount tree)
	   (3) major:minor:  value of st_dev for files on filesystem
	   (4) root:  root of the mount within the filesystem
	   (5) mount point:  mount point relative to the process's root
	   (6) mount options:  per mount options
	   (7) optional fields:  zero or more fields of the form "tag[:value]"
	   (8) separator:  marks the end of the optional fields
	   (9) filesystem type:  name of filesystem of the form "type[.subtype]"
	   (10) mount source:  filesystem specific information or "none"
	   (11) super options:  per super block options*/
```

**Comment by superseb** [comment]
LGTM

**Comment by StrongMonkey** [comment]
Fix https://github.com/rancher/rancher/issues/25506

File: main.go [hunk 1]
```diff
@@ -76,6 +76,17 @@ type Config struct {
 }
 
 func stage2(cli *cli.Context) (int, error) {
+	mounts, err := mount.GetMounts()
+	if err != nil {
+		return -1, err
+	}
+	for _, m := range mounts {
+		if m.Mountpoint == "/" && strings.HasPrefix(m.Optional, "shared") {
```
Comment (ibuildthecloud) [inline main.go]: i think this will have the format "tag:value tag:value".  I think you want to split (strings.Fields) and look for "shared:" prefix

Comment (ibuildthecloud) [reply main.go]: actually you need to after spliting the fields split on ":" and make sure the first part equals "shared"

Comment (StrongMonkey) [reply main.go]: addressed

**Comment by superseb** [comment]
LGTM

**Pull request merged**